lib/rex/post/meterpreter/extensions/stdapi/railgun/def/windows/def_wldap32.rb

```diff
@@ -12,111 +12,111 @@ class Def_windows_wldap32
     ], 'ldap_bind_sA', "cdecl")
 
     dll.add_function('ldap_search_sA', 'DWORD',[
-        ['DWORD', 'ld', 'in'],
+        ['LPVOID', 'ld', 'in'],
         ['PCHAR', 'base', 'in'],
-        ['DWORD', 'scope', 'in'],
+        ['ULONG', 'scope', 'in'],
         ['PCHAR', 'filter', 'in'],
         ['PCHAR', 'attrs[]', 'in'],
-        ['DWORD', 'attrsonly', 'in'],
-        ['PDWORD', 'res', 'out']
+        ['ULONG', 'attrsonly', 'in'],
+        ['PLPVOID', 'res', 'out']
     ], 'ldap_search_sA', "cdecl")
 
     dll.add_function('ldap_set_option', 'DWORD',[
-        ['DWORD', 'ld', 'in'],
+        ['LPVOID', 'ld', 'in'],
         ['DWORD', 'option', 'in'],
-        ['PDWORD', 'invalue', 'in']
+        ['PBLOB', 'invalue', 'in']
     ], 'ldap_set_option', "cdecl")
 
     dll.add_function('ldap_search_ext_sA', 'DWORD',[
-        ['DWORD', 'ld', 'in'],
+        ['LPVOID', 'ld', 'in'],
         ['PCHAR', 'base', 'in'],
-        ['DWORD', 'scope', 'in'],
+        ['ULONG', 'scope', 'in'],
         ['PCHAR', 'filter', 'in'],
         ['PCHAR', 'attrs[]', 'in'],
-        ['DWORD', 'attrsonly', 'in'],
-        ['DWORD', 'pServerControls', 'in'],
-        ['DWORD', 'pClientControls', 'in'],
-        ['DWORD', 'pTimeout', 'in'],
-        ['DWORD', 'SizeLimit', 'in'],
-        ['PDWORD', 'res', 'out']
+        ['ULONG', 'attrsonly', 'in'],
+        ['LPVOID', 'pServerControls', 'in'],
+        ['LPVOID', 'pClientControls', 'in'],
+        ['PBLOB', 'pTimeout', 'in'],
+        ['ULONG', 'SizeLimit', 'in'],
+        ['PLPVOID', 'res', 'out']
     ], 'ldap_search_ext_sA', "cdecl")
 
     dll.add_function('ldap_count_entries', 'DWORD',[
-        ['DWORD', 'ld', 'in'],
-        ['DWORD', 'res', 'in']
+        ['LPVOID', 'ld', 'in'],
+        ['LPVOID', 'res', 'in']
     ], "ldap_count_entries", "cdecl")
 
-    dll.add_function('ldap_first_entry', 'DWORD',[
-        ['DWORD', 'ld', 'in'],
-        ['DWORD', 'res', 'in']
+    dll.add_function('ldap_first_entry', 'LPVOID',[
+        ['LPVOID', 'ld', 'in'],
+        ['LPVOID', 'res', 'in']
     ], 'ldap_first_entry', "cdecl")
 
-    dll.add_function('ldap_next_entry', 'DWORD',[
-        ['DWORD', 'ld', 'in'],
-        ['DWORD', 'entry', 'in']
+    dll.add_function('ldap_next_entry', 'LPVOID',[
+        ['LPVOID', 'ld', 'in'],
+        ['LPVOID', 'entry', 'in']
     ], 'ldap_next_entry', "cdecl")
 
-    dll.add_function('ldap_first_attributeA', 'DWORD',[
-        ['DWORD', 'ld', 'in'],
-        ['DWORD', 'entry', 'in'],
-        ['DWORD', 'ptr', 'in']
+    dll.add_function('ldap_first_attributeA', 'PCHAR',[
+        ['LPVOID', 'ld', 'in'],
+        ['LPVOID', 'entry', 'in'],
+        ['PLPVOID', 'ptr', 'out']
     ], 'ldap_first_attributeA', "cdecl")
 
-    dll.add_function('ldap_next_attributeA', 'DWORD',[
+    dll.add_function('ldap_next_attributeA', 'PCHAR',[
         ['DWORD', 'ld', 'in'],
         ['DWORD', 'entry', 'in'],
-        ['DWORD', 'ptr', 'inout']
+        ['PBLOB', 'ptr', 'inout']
```

Comment:
I think the types of the `ptr` argument to both of these functions should be the same. Since we don't have a pointer to a PBLOB type (`PPBLOB`) I think it makes sense to use `LPVOID` for the second. That would mean the caller would need to manage the memory themselves and would keep the datatype the same here for the parameter (integer for LPVOID vs changing it to a string for PBLOB as it is now.

Reply:
So keep the first one as PLPVOID to reflect the `BerElement  **ptr` part and then update the second to be a `LPVOID` pointer to reflect the `BerElement  *ptr` part? Sounds good to me.

Reply:
Should be fixed with 39651f46828511742d0c469a1e72a957ea8eb2b6 but let me know if this needs additional work.